src/Language/C99/Simple/Translate.hs

```diff
@@ -253,11 +275,27 @@ assignop op e1 e2 = C.Assign e1' op' e2' where
 
 transinit :: Init -> C.Init
 transinit (InitExpr e)   = C.InitExpr (wrap $ transexpr e)
-transinit (InitArray es) = C.InitArray (fromList $ map transinit es)
-
-initexpr ty init = C.PostfixInits ty' init' where
-  ty'   = transtypename ty
-  init' = fromList $ map transinit init
```

Comment:
Got rid of some uses of `fromList`. The relevant `IsList` instances are in `language-c99-util` and are partial. Ideally we'd drop all use of them, as IMO they are quite confusing. Same goes for the `wrap` motif.